lib/scholar/metrics/distance.ex

```diff
@@ -407,4 +426,61 @@ defmodule Scholar.Metrics.Distance do
     res = Nx.select(one_zero?, 0.0, res)
     1.0 - Nx.select(both_zero?, 1.0, res)
   end
+
+  @doc """
+  Hamming distance.
+
+  N_unequal(x, y) / N_tot
+
+  ## Options
+
+  #{NimbleOptions.docs(@hamming_schema)}
+
+  ## Examples
+
+      iex> x = Nx.tensor([1, 0, 0])
+      iex> y = Nx.tensor([0, 1, 0])
+      iex> Scholar.Metrics.Distance.hamming(x, y)
+      #Nx.Tensor<
+        f32
+        0.6666666865348816
+      >
+      iex> Scholar.Metrics.Distance.hamming(x, y, weights: [1,0.5,0.5])
+      #Nx.Tensor<
+        f32
+        0.75
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2, 3])
+      iex> Scholar.Metrics.Distance.hamming(x, y)
+      ** (ArgumentError) expected tensor to have shape {2}, got tensor with shape {3}
+
+      iex> x = Nx.tensor([[1, 2, 3], [0, 0, 0], [5, 2, 4]])
+      iex> y = Nx.tensor([[1, 5, 2], [2, 4, 1], [0, 0, 0]])
+      iex> Scholar.Metrics.Distance.hamming(x, y, axis: 1)
+      #Nx.Tensor<
+        f32[3]
+        [0.6666666865348816, 1.0, 1.0]
+      >
+  """
+  deftransform hamming(x, y, opts \\ []) do
+    options = NimbleOptions.validate!(opts, @hamming_schema)
+
+    if nil == options[:weights] do
+      hamming_unweighted(x, y, options)
+    else
+      hamming_weighted(x, y, options)
+    end
+  end
+
+  defnp hamming_unweighted(x, y, opts \\ []) do
+    assert_same_shape!(x, y)
+    (x != y) |> Nx.mean(axes: [opts[:axis]])
```

Comment:
Why can't we support :axes instead of just a single axis?

Reply:
This was mainly due to `weighted_mean` only taking `axis` as a param. This brings up additional questions as to whether we'd want to support multi-dim tensors (scipy elects to only support 1D). The current `moduledoc` for `Metrics.Distance` reads `Distance metrics between 1-D tensors.`

Reply:
Perhaps weighted_mean could be extended for having N-dimensional weights?

Reply:
@seanmor5 I think this is more of a question for you

Reply:
It was kind of a self-question as well haha!

Reply:
`Nx.mean` accepts `axes` so it seems like it would be consistent to have it for `weighted_mean` as well

Reply:
@acalejos https://github.com/elixir-nx/nx/pull/1043 adds support for `:axes` instead of `:axis` in `Nx.weighted_mean`

Reply:
Oh nice. Should this wait on that getting merged into Nx then?

Reply:
You can point to my branch while it's not merged so you can keep working on it :)

Reply:
What do you mean exactly by point to your branch? Would I do that from the mix.exs file?

Reply:
Yes!

You can change the nx declaration in mix.exs to:

`{:nx, ">= 0.0.0", github: "elixir-nx/nx", sparse: "nx", branch: "pv-feat/multi-axes-weighted-mean"}`

This is basically saying: fetch from the elixir-nx/nx repo, only the nx directory, and checkout the given branch